test/integration/podgc/podgc_test.go

```diff
@@ -28,152 +28,238 @@ import (
+					Name: "node",
+				},
+				Spec: v1.NodeSpec{},
+				Status: v1.NodeStatus{
+					Conditions: []v1.NodeCondition{
+						{
+							Type:   v1.NodeReady,
+							Status: v1.ConditionTrue,
+						},
+					},
+				},
+			}
+			node, err := cs.CoreV1().Nodes().Create(testCtx.Ctx, node, metav1.CreateOptions{})
+			if err != nil {
+				t.Fatalf("Failed to create node '%v', err: %v", node.Name, err)
+			}
 
-	// we delete the node to orphan the pod
-	err = cs.CoreV1().Nodes().Delete(testCtx.Ctx, pod.Spec.NodeName, metav1.DeleteOptions{})
-	if err != nil {
-		t.Fatalf("Failed to delete node: %v, err: %v", pod.Spec.NodeName, err)
-	}
+			pod := &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:       "testpod",
+					Namespace:  testCtx.NS.Name,
+					Finalizers: []string{"test.k8s.io/finalizer"},
+				},
+				Spec: v1.PodSpec{
+					NodeName: node.Name,
+					Containers: []v1.Container{
+						{Name: "foo", Image: "bar"},
+					},
+				},
+			}
 
-	err = wait.PollImmediate(time.Second, time.Second*15, func() (bool, error) {
-		updatedPod, err := cs.CoreV1().Pods(testCtx.NS.Name).Get(testCtx.Ctx, pod.Name, metav1.GetOptions{})
-		if err != nil {
-			return true, err
-		}
-		if updatedPod.ObjectMeta.DeletionTimestamp != nil {
-			return true, nil
-		}
-		return false, nil
-	})
-	if err != nil {
-		t.Fatalf("Error '%v' while waiting for the pod '%v' to be deleted", err, klog.KObj(pod))
+			pod, err = cs.CoreV1().Pods(testCtx.NS.Name).Create(testCtx.Ctx, pod, metav1.CreateOptions{})
+			if err != nil {
+				t.Fatalf("Error %v, while creating pod: %v", err, klog.KObj(pod))
+			}
+			defer testutils.RemovePodFinalizers(testCtx.ClientSet, t, []*v1.Pod{pod})
+
+			// we delete the node to orphan the pod
+			err = cs.CoreV1().Nodes().Delete(testCtx.Ctx, pod.Spec.NodeName, metav1.DeleteOptions{})
+			if err != nil {
+				t.Fatalf("Failed to delete node: %v, err: %v", pod.Spec.NodeName, err)
+			}
+			err = wait.PollImmediate(time.Second, time.Second*15, testutils.PodIsGettingEvicted(cs, pod.Namespace, pod.Name))
+			if err != nil {
+				t.Fatalf("Error '%v' while waiting for the pod '%v' to be terminating", err, klog.KObj(pod))
+			}
+			pod, err = cs.CoreV1().Pods(testCtx.NS.Name).Get(testCtx.Ctx, pod.Name, metav1.GetOptions{})
+			if err != nil {
+				t.Fatalf("Error: '%v' while updating pod info: '%v'", err, klog.KObj(pod))
+			}
+			_, cond := podutil.GetPodCondition(&pod.Status, targetedForDisruption)
+			if test.enableJobPodFailurePolicy == true && cond == nil {
+				t.Fatalf("Pod %v/%v does not have the expected condition: %v", pod.Namespace, pod.Name, targetedForDisruption)
+			} else if test.enableJobPodFailurePolicy == false && cond != nil {
+				t.Fatalf("Pod %v/%v does has the unexpected condition: %v", pod.Namespace, pod.Name, targetedForDisruption)
+			}
+			if test.wantPhase != pod.Status.Phase {
+				t.Fatalf("Enexpected phase for pod '%v'. Expected: '%v;, got: '%v'", klog.KObj(pod), test.wantPhase, pod.Status.Phase)
```

Comment:
Prefer `Errorf` when checking the expectations of the test. This allows the test to continue and get the remaining errors.

`Fatalf` makes sense when the test setup fails, because there is no point continuing.

```suggestion
				t.Errorf("Unexpected phase for pod %q. Got %q, want %q'", klog.KObj(pod), pod.Status.Phase, test.wantPhase)
```